Block/NodeType/Category.php

```diff
@@ -102,7 +110,7 @@ public function fetchData(array $nodes)
     {
         $storeId = $this->_storeManager->getStore()->getId();
 
-        list($this->nodes, $this->categoryUrls) = $this->_categoryModel->fetchData($nodes, $storeId);
+        list($this->nodes, $this->categoryUrls, $this->categories) = $this->_categoryModel->fetchData($nodes, $storeId);
```

Comment:
The `$this->categoryUrls` property becomes obsolete if the rest of the changes are approved. I left it in until I've gotten some feedback on whether my approach is acceptable in the first place.